src/hooks/useColumns.test.ts

```
@@ -0,0 +1,61 @@
+import { renderHook } from '@testing-library/react';
+import { act } from 'react-dom/test-utils';
+import { Datasource } from 'data/CHDatasource';
+import useColumns from './useColumns';
+import { TableColumn } from 'types/queryBuilder';
+
+describe('useColumns', () => {
```

Should we add a test that an exception is not swallowed?

---

The `throw ex;` syntax you pointed out in the source file was indeed being lost. Right now the only way to see that error would be to check the console. If the fetch fails then the array is left empty. Could add a case for it